### ISSUE TITLE
VT logs not unicode safe (critical)

### ISSUE THREAD

**Issue by kiorky** [saltstack/salt#21441]
to reproduce the bug

```
salt-call --hard-crash --local  -lall cmd.run 'for i in $(seq 2);do echo $i;sleep 1;done;echo é'  
python_shell=True use_vt=True
```

reopening of unfixed #20918

**Comment by rallytime** [comment]
Thanks for filing this issue, @kiorky. You're running into this issue on the develop branch, correct?

**Comment by kiorky** [comment]
our fork is more close to develop than 2015.2 but i think this should also affect 2015.2 as vt is in.

**Comment by kiorky** [comment]
(It's not a blocker for me, as my workaround did unblock us since).

**Comment by thatch45** [comment]
Yes, this is an issue that we need to get fixed deeper then it was in #20918 is all, thanks @kiorky

**Comment by s0undt3ch** [comment]
Proper Unicode support is going to be a massive task and we shouldn't just be adding band-aids to the code, it will only fix some spots of Salt's code base.

The start of it has begun on 2015.2 but it's only to detect the system encoding, any further developments around Unicode should and will happen on the develop branch as soon as we're confident that the detection works as supposed.

So, while this is valid bug report, it's related to a deeper enhancement in Salt.
Let's keep this open(or at most, open a new one, Unicode support related) so we can keep track of further changes to the underlining issue.
